third-party/crowdsec/src/test/java/org/apache/james/CrowdsecExtension.java

```diff
@@ -0,0 +1,81 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import java.time.Duration;
+import java.util.UUID;
+
+import org.apache.james.util.docker.RateLimiters;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.HostPortWaitStrategy;
+
+public class CrowdsecExtension implements BeforeAllCallback {
+    public static final Duration STARTUP_TIMEOUT = Duration.ofMinutes(5);
+    public static final int CROWDSEC_PORT = 8080;
+    public static final String CROWDSEC_IMAGE = "crowdsecurity/crowdsec:v1.5.4";
+    private static final GenericContainer<?> crowdsecContainer = new GenericContainer<>(CROWDSEC_IMAGE)
```

Comment:
Would prefer declaring the container in a create method...

Honestly overall look at other examples... `RspamdExtension` for example